File: app/src/main/java/com/dominikgaller/alpinebooking/shared/domain/event/TourStarted.java
package com.dominikgaller.alpinebooking.shared.domain.event;

import com.dominikgaller.alpinebooking.shared.domain.TourId;

import java.time.Instant;

/**
 * Integration event emitted by the {@code guide} bounded context when a guide tour is started.
 *
 * <p>Lives in {@code shared.domain.event} so the {@code booking} context can subscribe
 * without depending on the {@code guide} package. {@code guideTourId} is carried as a
 * plain {@link String} to avoid a {@code shared → guide} compile-time dependency.
 *
 * <p>Immutable. Published after the guide tour's transaction commits (see ADR 0002).
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}, section 6,
 *          and {@code documentation/adr/0005-bounded-context-identity-boundaries.adr.md}
 *          for the foreign-identity rule.
 */
public record TourStarted(
        String guideTourId,
        TourId tourId,
        Instant startedAt
) implements DomainEvent {
}

File: app/src/main/kotlin/com/example/service/shared/domain/event/TourStarted.kt
package com.example.service.shared.domain.event

import com.example.service.shared.domain.TourId
import java.time.Instant

/**
 * Integration event emitted by the `guide` bounded context when a guide tour is started.
 *
 * Lives in `shared.domain.event` so the `booking` context can subscribe without depending
 * on the `guide` package. `guideTourId` is carried as a plain [String] to avoid a
 * `shared -> guide` compile-time dependency.
 *
 * Immutable. Published after the guide tour's transaction commits (ADR 0002).
 *
 * SDD: see `documentation/use-cases/uc05-start-tour.spec.md` section 6, and
 * `documentation/adr/0005-bounded-context-identity-boundaries.adr.md` for the
 * foreign-identity rule.
 */
data class TourStarted(
    val guideTourId: String,
    val tourId: TourId,
    val startedAt: Instant,
) : DomainEvent
